src/olympia/reviews/views.py

```diff
@@ -495,3 +499,22 @@ def flag(self, request, *args, **kwargs):
             response.content = ''
             response.status_code = 202
         return response
+
+    @detail_route(methods=['post'])
+    def delete(self, request, *args, **kwargs):
```

Comment:
I'd prefer a refactor to the model to remove `moderator_delete()` entirely, moving the `not review.addon.has_author(request.user)` check inside the `Review.delete()` method. Then, in the API all you need to do is remove that custom `delete()` method and instead add this:
```python
def perform_destroy(self, instance):
    instance.delete(user_responsible=self.request.user)
```

That avoid the copy/paste from the flag method you did below and keeps core stuff in the model code.